clang/lib/Sema/SemaTemplate.cpp

```diff
@@ -2653,20 +2653,34 @@ struct ConvertConstructorToDeductionGuideTransform {
 // Find all template parameters that appear in the given DeducedArgs.
 // Return the indices of the template parameters in the TemplateParams.
 SmallVector<unsigned> TemplateParamsReferencedInTemplateArgumentList(
-    ArrayRef<NamedDecl *> TemplateParams,
+    const TemplateParameterList* TemplateParamsList,
```

Comment:
The template parameters themselves should also have a depth, not just the index.
So you don't need to change everything to pass a TemplateParameterList instead of the array as before.

In fact, TemplateParameterList will just return the depth of its first parameter.

Reply:
> In fact, TemplateParameterList will just return the depth of its first parameter.

I think it is fine, my understanding is that all parameters within a template parameter list should have the same depth,